app/assets/javascripts/controllers/middleware_deployment/middleware_deployment_controller.js

```diff
@@ -0,0 +1,38 @@
+ManageIQ.angular.app.controller('middlewareAddDeploymentController', MiddlewareDeploymentCtrl);
+
+MiddlewareDeploymentCtrl.$inject = ['$scope', '$http', 'miqService'];
+
+function MiddlewareDeploymentCtrl($scope, $http, miqService) {
+  ManageIQ.angular.scope = $scope;
+
+  $scope.showListener = function () {
+    $scope.showDeployModal = true;
+  };
+
+  $scope.addDeployment = function () {
+    miqService.sparkleOn();
+    var url = '/middleware_server/add_deployment';
+    $scope.uploadFile($scope.filePath, url);
+  };
+    
+  $scope.uploadFile = function (file, uploadUrl) {
+    var fd = new FormData();
+    fd.append('file', file);
+    fd.append('id', angular.element('#server_id').val());
+    $http.post(uploadUrl, fd, {
+      transformRequest: angular.identity,
+      headers: {'Content-Type': undefined}
+    })
+    .success(function() {
```

Comment:
`$http.VERB.success` and `.error` are deprecated use `then` instead.

From https://docs.angularjs.org/api/ng/service/$http: "The $http legacy promise methods success and error have been deprecated. Use the standard then method instead. If $httpProvider.useLegacyPromiseExtensions is set to false then these methods will throw $http/legacy error."